components/director/internal/formationmapping/handler.go

```diff
@@ -393,8 +393,10 @@ func (h *Handler) processFormationUnassign(ctx context.Context, formation *model
 		return errors.Wrapf(err, "while unassigning object with type: %q and ID: %q", fa.SourceType, fa.Source)
 	}
 
-	if err = h.unassignObjectFromFormationWhenThereAreNoFormationAssignments(unassignCtx, fa, formation, fa.Target, fa.TargetType); err != nil {
-		return errors.Wrapf(err, "while unassigning object with type: %q and ID: %q", fa.TargetType, fa.Target)
+	if fa.Source != fa.Target {
```

Comment:
Do you think a comment with info will be helpful here?
e.g.
```
// Skip second unassign from formation in case the formation assignment is self-referenced one, 
// and if happened to be the last one. It will be handled above.
```

Reply:
Yes, it probably will be useful, I'll add it.